compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml

```diff
@@ -175,3 +179,30 @@ class HasInterfaceView i v | i -> v where
 
 view : forall i v. HasInterfaceView i v => i -> v
 view = _view
+
+class HasFromAnyView i v | i -> v where
+  -- | HIDE
+  _fromAnyView : proxy i -> Any -> Optional v
+
+class HasToAnyView i v | i -> v where
+  -- | HIDE
+  _toAnyView : proxy i -> v -> Any
+
+fromAnyView : forall i v. (HasTemplateTypeRep i, HasFromAnyView i v) => AnyView -> Optional v
+fromAnyView (AnyView any typeRep)
+  | Some c <- _fromAnyView ([] : [i]) any
+  , fromTemplateTypeRep (templateTypeRep @i) == typeRep = Some c
+  | otherwise = None
+
+toAnyView : forall i v. (HasTemplateTypeRep i, HasToAnyView i v) => v -> AnyView
+toAnyView v =
+  AnyView
+    (_toAnyView ([] : [i]) v)
+    (fromTemplateTypeRep (templateTypeRep @i))
+
+deriving instance Eq InterfaceTypeRep
+deriving instance Ord InterfaceTypeRep
```

Comment:
Following from [this comment](https://github.com/digital-asset/daml/pull/15280/files#r1003072116), these instances would go under `DA.Internal.Interface.AnyView`

Reply:
We should also add the appropriate `HasField` instances under `DA.Internal.Interface.AnyView`

Reply:
done